labs/lab03.md

```diff
@@ -0,0 +1,52 @@
+# Lab 3: Testing
+
+In this lab you will get familiar with different testing strategies and practice with writing good unit tests in Java and TypeScript.
+
+## Concepts
+**Structural testing vs Specification testing:** Two distinct types of unit testing are specification-based (historically: black-box) testing and structural (white-box) testing. Specification testing is when you write high-level test cases against the specification of a program, without using (or testing) the specific underlying implementation of the program. In structural testing, your tests are instead tailored to the specific underlying implementation of the code you are testing.
+
+**Code coverage:** Coverage is the amount of code that has been executed by the tests. There are several coverage metrics, but one common metric line coverage is simply the percentage of lines (or statements) that have been executed by your unit tests. 
+
+## Deliverables
+
+- [ ] Write tests for LinkedIntQueue in TypeScript and ArrayIntQueue in Java and execute them from within the IDE.
+- [ ] Achieve 100% line coverage for ArrayIntQueue and show coverage results in the IDE or as HTML report. 
+- [ ] Compare structural testing vs specification testing and analyze tradeoffs. 
+
+
+## Instructions
+
+### Setup
+Fork and clone the repo from https://github.com/CMU-17-214/f23-rec03.
```

Comment:
can you check whether we can or should update any dependencies?

For TS we have started using a different module configuration (see tsconfig file in lab2 and hw1). Probably doesn't matter but might be nice to be consistent here and avoid surprises for hw2

Reply:
Sure! I've copied tsconfig file in lab2 and updated packages: https://github.com/CMU-17-214/f2023/pull/1